sei/on_poll.py

```diff
@@ -1,21 +1,22 @@
 import phantom.app as phantom
-
-from .get_events import get_events
+from phantom.action_result import ActionResult
+import sei
 
 def on_poll(connector, param):
-    timestamp = connector._state["last_poll"] if "last_poll" in connector._state.keys() else '2022-08-01T00:00:01Z'
+    timestamp = connector._timestamp if connector._timestamp != None else connector._state["last_poll"] if "last_poll" in connector._state.keys()  else '2022-08-01T00:00:01Z'
     container_id = connector._state["container_id"] if "container_id" in connector._state.keys() else None
-    
+    print(timestamp)
     if container_id is None or connector.get_container_info(container_id)[0] == False:
         container = {'name': 'WithSecure - security events', 'label': 'events',
-                     'container_type': 'default'}
+                     'container_type': 'default', "run_automation": True}
         success, message, container_id = connector.save_container(container, fail_on_duplicate=True)
 
-
-    artifacts, new_timestamp, action_result = get_events(connector, param, container_id, timestamp)
+    action_result = connector.add_action_result(ActionResult(dict(param)))
+    artifacts, new_timestamp, action_result = sei.get_events(connector, action_result, container_id, timestamp)
     if len(artifacts) > 0:
             success, message, id_list = connector.save_artifacts(artifacts)
 
+    connector._state["isolated_devices"] = []
```

Comment:
Every time on_poll action is executed it will clear all isolated devices cached in _state. As I understand this is to prevent multiple isolate_device actions for single device after polling action but with next poll it will be possible to isolate previously isolated device again. Is this intended behaviour or a mistake?

Reply:
When polling 100 events:

- Obtain their data.
- Isolate devices from events if necessary.

However, there could be a scenario where the same device appears multiple times within the 100 events. The isolation of a device is only maintained during one poll because, in the meantime, this device may have already been released from isolation. Therefore, we don't want to store the UUID of this device for longer than one poll.